controllers/controllers.go

```diff
@@ -132,6 +136,90 @@ func (r *instanceReconciler) instanceFromNode(node *core.Node) (*instance.Info,
 	return instance.NewInfo(addr, username, "", false, node)
 }
 
+// reconcileKubeletClientCA reconcile the CA certificate across all Windows nodes required by kubelet to recognize
+// the kube-apiserver client. No drain or restart required.
+func (r *instanceReconciler) reconcileKubeletClientCA(ctx context.Context, initialCAConfigMap,
+	bundleCAConfigMap *core.ConfigMap) error {
+	r.log.Info("reconciling kubelet CA client certificates with", "ConfigMap", bundleCAConfigMap.Name)
+	// extract CAs form initial CA ConfigMap
+	initialKubeAPIServerServingCABytes, err := certificates.GetCAsFromConfigMap(initialCAConfigMap,
+		certificates.CABundleKey)
+	if err != nil {
+		return err
+	}
+	// extract CAs from new bundle CA ConfigMap
+	kubeAPIServerServingCABytes, err := certificates.GetCAsFromConfigMap(bundleCAConfigMap, certificates.CABundleKey)
+	if err != nil {
+		return err
+	}
+	// merge initial certificates with new bundle
+	kubeAPIServerServingCABytes = certificates.MergeCertWithCABundle(initialKubeAPIServerServingCABytes,
+		kubeAPIServerServingCABytes, "kube-apiserver-to-kubelet-signer")
+
+	// TODO: Improve windows.sshConnectivity.transfer() to avoid below IO operations.
+	// We already have the content of the the bundle CA in memory, but we are forced to dump it into a file because
+	// is the only mechanism accepted by windows.sshConnectivity.transfer(). The later IO operations are expensive and
+	// could be avoided by overloading .transfer() to accept a byte array.
+
+	// create auxiliary file to write the CA certificate content
+	kubeletCAFilePath := filepath.Join(os.TempDir(), windows.KubeletClientCAFilename)
+	r.log.V(1).Info("creating auxiliary kubelet CA certificate", "file", kubeletCAFilePath)
+	kubeletCAFilePath, err = createFile(kubeletCAFilePath, kubeAPIServerServingCABytes)
+	if err != nil {
+		return errors.Wrapf(err, "error creating file: %s", kubeletCAFilePath)
+	}
+	// cleanup auxiliary certificate file
+	defer func(name string) {
+		r.log.V(1).Info("deleting kubelet CA certificate", "file", name)
+		if err := os.Remove(name); err != nil {
+			r.log.V(1).Error(err, "error deleting kubelet CA certificate", "file", name)
+		}
+	}(kubeletCAFilePath)
+
+	// fetch all Windows nodes (Machine and BYOH instances)
+	winNodes := &core.NodeList{}
+	if err = r.client.List(ctx, winNodes, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return errors.Wrap(err, "error listing Windows nodes")
+	}
+	r.log.V(1).Info("processing", "node count", len(winNodes.Items))
+	// loop Windows nodes and trigger kubelet CA update
+	for _, winNode := range winNodes.Items {
```

Comment:
Assume if we have 3 Windows nodes and the second one fails due to some error condition, aren't we going to try the 3rd node?

Reply:
Good question. In case the 3rd node fails, the reconciler will reconcile the failed event where nodes 1 and 2 are up to date, and only the 3rd node will get the file update.